dataservice/api/errors.py

```diff
@@ -1,6 +1,45 @@
 from dataservice.api.common.schemas import ErrorSchema
 
+FOREIGN_KEY_ERROR_PREFIX = 'foreign key constraint failed'
+FOREIGN_KEY_ERROR_MSG_TEMPLATE = \
+    'Cannot {} {} without an existing {} entity'
+
+UNIQUE_CONSTRAINT_ERROR_PREFIX = 'unique constraint failed'
+UNIQUE_CONSTRAINT_ERROR_MSG_TEMPLATE = \
+    'Cannot {0} {1}, {2} already has a {1}'
+
+DEFAULT_INVALID_INPUT_TEMPLATE = 'Client error: invalid {} provided'
+
 
 def http_error(e):
     """ Handles all HTTPExceptions """
     return ErrorSchema().jsonify(e), e.code
+
+
+def handle_integrity_error(**kwargs):
```

Comment:
All of this shouldn't be necessary when we move to postgres. It returns much more detailed exceptions that can be parsed out in the native handler rather than an intermediate parser.

Reply:
👍